src/main/java/unbanner/BuildingController.java

```diff
@@ -140,6 +144,7 @@ public String newRoom(@ModelAttribute("building") Building building,
     newRoom.building = thisBuilding;
     thisBuilding.rooms.add(newRoom);
     repository.save(thisBuilding);
+    roomRepository.save(newRoom);
```

Comment:
need to check for conflicts before saving this room and building within this method, Or else this allows the creation of new rooms with the same name as already existing rooms.